### DISCUSSION TITLE
Curating the organisational Github presence

### DISCUSSION THREAD

**Discussion by metazool** [NERC-CEH/rse_group#44] [Ideas]
Wanted to start a thread for @jmarshrossney 's post-RSECon suggestion, as it's a good one - simply actionable, and helps build internal community.

## Suggesting actions to the group owners

* https://github.com/NERC-CEH/ - identify "flagship" projects that can be pinned. 
* Send examples organisations curating their presence well - https://github.com/antarctica is a nice related one

## Suggesting actions to project maintainers

* Encourage people to explicitly label experimental projects as such
* Encourage them to suggest their own work be pinned or promoted!

## Actions for ourselves 

* Find Github organisations that are used by specific research groups or projects outwith the central one, add them here

**Comment by jmarshrossney** [comment]
What do we think about trying to reach a consensus (with the wider EDS group) on a set of guidelines for what should be put on the organisational GitHub versus personal ones, perhaps including a checklist "have you done these things" before you publish a repo on the organisational GitHub?

I worry this might annoy some people if they feel targeted, but it could also be productive to prompt people to revisit some of their old repos and decide whether to archive them, document them slightly better, or move them to their personal accounts. 

At the very least some clear guidance for new repos seems like a good idea.

**Reply by longr** [reply]
I think one issue is that it is good practise to start projects in the organisational one so that anyone can work on it/others have access if needed.   Perhaps tagging as production ready/in development/new_project might be more useful.

**Reply by jmarshrossney** [reply]
I can definitely see the value in that for projects, but people use GitHub for all kinds of 'non-project' stuff, like personal scripts, presentations, tests etc. For me these would be better off on a user account because the chance of them being useful to a random member of UKCEH browsing through repositories is slim, and they make it harder to find the projects.

Remember we are trying to persuade many people to use GitHub who currently aren't - so if our plan works the number of these kinds of repositories might increase 5x!

**Comment by metazool** [comment]
https://github.com/BioDT/ has an organisation here (Biodiversity Digital Twins)

**Reply by jmarshrossney** [reply]
This is great!

All of the repositories associated with a project in one place, with an informative README!

It would be nice if NERC-CEH had a landing page / README that listed active and past projects and linked to their organisational GitHub pages.

**Comment by metazool** [comment]
@dcmo you might be interested in this too - while we're trying to identify and suggest incremental improvements to the organisational Github presence, also trying to get a sense of who's responsible enough for it to [answer to admin requests](https://github.com/orgs/NERC-CEH/people?query=role%3Aowner) - and whether our RSE group should be stepping up to shoulder responsibility with @cehscicom...

**Reply by DamienUKCEH** [reply]
yes am fine with curating the repositories

**Comment by longr** [comment]
Tagging in @jez-carter from the Data Science Team.  Jez was looking at what the DS team could do, and whether they need a different GH Org to manage just DS repos, so thought it would be good to connect and look at what does and doesn't work with GH org at the moment.

**Reply by Jez-Carter** [reply]
Thanks @longr, yes I was interested in whether it makes sense to have subgroups of CEH as organisations in GitHub also, as for example the RSE team has for Newcastle University (https://github.com/NewcastleRSE). The reason I started thinking down these lines is that I imagine that the Nerc-CEH will easily become cluttered with repos and I also prefer the idea of managing the DS team's GitHub organisation within our own team (e.g. having the group lead and at least one other as owners on the sub-group organisation). Still need to think about this and whether it would really be that beneficial, especially if there is progress with improving the Nerc-CEH organisation page.

**Reply by jmarshrossney** [reply]
@Jez-Carter I'm with you 100% on this. NERC-CEH can probably do without random presentations, team organisation stuff etc.

I think there's a valid debate around discoverability / ease of 'dropping in' on a piece of work that's not led by your own team. It's nice to have an overview of what's currently being worked on based on the active repositories on NERC-CEH. 

But I would argue that not everything needs to compete with projects in that space; not everything needs to be that readily discoverable by the wider organisation!

To make a more distributed approach work well, we could:
- Make sure the NERC-CEH README.md clearly lists affiliated organisations, including those related to projects/collaborations such as https://github.com/BioDT/
- Develop some simple, clear guidance on 'where does your repository belong'. 
- Put effort into fostering better communication and 'sharing of updates' between members of the org (much better than basing one's picture of what's being worked on on the NERC-CEH's recently updated repositories)

**Comment by jmarshrossney** [comment]
This is probably a good time to flag up that I've gone and created a trial GitHub org for the RSE group here: https://github.com/ukceh-rse , to avoid duplication of effort.

I'm not 100% sure that I have the 'authority' to use the UKCEH brand in this way, which is why its description is currently

> This is JUST AN EXPERIMENT to see whether we want to bother with a separate GitHub Organisation for our RSE team

My intention was to use Thursday's catch up to make an argument in favour of an organisation instead of [a repository](https://github.com/NERC-CEH/rse_group) , using some concrete examples.

@Jez-Carter you'd be welcome to join if this discussion would be useful to you / data science. I'm sure there will be some robust counter arguments!

**Reply by Jez-Carter** [reply]
Thanks, yeah I'd be interested to join, could you forward me the meeting invite?
